examples/baseball.py

```diff
@@ -235,78 +257,95 @@ def main(args):
     at_bats, hits = train[:, 0], train[:, 1]
     logging.info("Original Dataset:")
     logging.info(baseball_dataset)
-    num_predictive_samples = args.num_samples * args.num_chains
 
     # (1) Full Pooling Model
-    nuts_kernel = NUTS(fully_pooled)
-    posterior_fully_pooled = MCMC(nuts_kernel,
-                                  num_samples=args.num_samples,
-                                  warmup_steps=args.warmup_steps,
-                                  num_chains=args.num_chains).run(at_bats, hits)
+    init_params, potential_fn, transforms, _ = initialize_model(fully_pooled, model_args=(at_bats, hits),
+                                                                num_chains=args.num_chains)
+    nuts_kernel = NUTS(potential_fn=potential_fn)
+    mcmc = MCMC(nuts_kernel,
+                num_samples=args.num_samples,
+                warmup_steps=args.warmup_steps,
+                num_chains=args.num_chains,
+                initial_params=init_params,
+                transforms=transforms)
+    num_chains = mcmc.num_chains  # Note that this might be less than args.num_chains  when CPU cores are limited.
+    samples_fully_pooled = mcmc.run(at_bats, hits)
```

Comment:
@fehiepsi - We will need to change all these examples / tests if we don't want `.run(..)` to return the dict of samples as discussed in #1942. I'm happy to do that in a separate PR.